tools/src/test/java/org/apache/kafka/tools/consumer/group/DescribeConsumerGroupTest.java

```diff
@@ -16,834 +16,986 @@
             cgcArgs.addAll(describeType);
-            ConsumerGroupCommand.ConsumerGroupService service = getConsumerGroupService(cgcArgs.toArray(new String[0]));
-
-            String output = ToolsTestUtils.grabConsoleOutput(describeGroups(service));
-            assertTrue(output.contains("Consumer group '" + missingGroup + "' does not exist."),
-                "Expected error was not detected for describe option '" + String.join(" ", describeType) + "'");
+            try (ConsumerGroupCommand.ConsumerGroupService service = consumerGroupService(cgcArgs.toArray(new String[0]));) {
+                String output = ToolsTestUtils.grabConsoleOutput(describeGroups(service));
+                assertTrue(output.contains("Consumer group '" + missingGroup + "' does not exist."),
+                        "Expected error was not detected for describe option '" + String.join(" ", describeType) + "'");
+            }
         }
     }
 
-    @ParameterizedTest
-    @ValueSource(strings = {"zk", "kraft"})
-    public void testDescribeWithMultipleSubActions(String quorum) {
-        AtomicInteger exitStatus = new AtomicInteger(0);
-        AtomicReference<String> exitMessage = new AtomicReference<>("");
-        Exit.setExitProcedure((status, err) -> {
-            exitStatus.set(status);
-            exitMessage.set(err);
-            throw new RuntimeException();
-        });
-        String[] cgcArgs = new String[]{"--bootstrap-server", bootstrapServers(listenerName()), "--describe", "--group", GROUP, "--members", "--state"};
-        try {
-            assertThrows(RuntimeException.class, () -> ConsumerGroupCommand.main(cgcArgs));
-        } finally {
-            Exit.resetExitProcedure();
+    @ClusterTemplate("generator")
+    public void testDescribeOffsetsOfNonExistingGroup(ClusterInstance clusterInstance) throws Exception {
+        this.clusterInstance = clusterInstance;
+        String missingGroup = "missing.group";
+        for (GroupProtocol groupProtocol: clusterInstance.supportedGroupProtocols()) {
+            String topic = TOPIC_PREFIX + groupProtocol.name();
+            String group = GROUP_PREFIX + groupProtocol.name();
+            createTopic(topic);
+
+            // run one consumer in the group consuming from a single-partition topic
+            try (AutoCloseable protocolConsumerGroupExecutor = consumerGroupClosable(groupProtocol, group, topic, Collections.emptyMap());
+                 // note the group to be queried is a different (non-existing) group
+                 ConsumerGroupCommand.ConsumerGroupService service = consumerGroupService(new String[]{"--bootstrap-server", clusterInstance.bootstrapServers(), "--describe", "--group", missingGroup});
+            ) {
+                Entry<Optional<ConsumerGroupState>, Optional<Collection<PartitionAssignmentState>>> res = service.collectGroupOffsets(missingGroup);
```

Comment:
please remove this unused variable

Reply:
Hi @chia7712, The `res` value is used by next line for assertion. It's not unused variable. Thanks.